entities/entities/darkrp_laws/cl_init.lua

```diff
@@ -33,26 +33,21 @@ end
 local function addLaw(inLaw)
     local law = DarkRP.textWrap(inLaw, "TargetID", 522)
 
-    Laws[#Laws + 1] = law
+    table.insert( Laws, law )
+    hook.Run("addLaw", #Laws + 1, law)
 end
 
 local function umAddLaw(um)
     local law = um:ReadString()
     timer.Simple(0, fn.Curry(addLaw, 2)(law))
-    hook.Run("addLaw", #Laws + 1, law)
 end
 usermessage.Hook("DRP_AddLaw", umAddLaw)
 
 local function umRemoveLaw(um)
     local i = um:ReadShort()
 
+    table.remove( Laws, i )
     hook.Run("removeLaw", i, Laws[i])
```

Comment:
Similar situation here - `Laws[i]` will always return nil here. Save it to a local variable before removing.